### DISCUSSION TITLE
LOG_FAILED_LOGIN_ATTEMPTS for Fail2ban or?

### DISCUSSION THREAD

**Discussion by sideskift** [plausible/analytics#3366] [Self-Hosted Support]
Hi 

The LOG_FAILED_LOGIN_ATTEMPTS Configuration option, is that for Fail2ban or simular? 

If so, any tips or hint on how to implement it? The documentation does not have much information regarding this setting.

Cheers

**Comment by ruslandoga** [comment]
👋 @sideskift 

`LOG_FAILED_LOGIN_ATTEMPTS` only does what it says it does :)

When `LOG_FAILED_LOGIN_ATTEMPTS` is set to true, the failed login attempts start showing up in `plausible` container logs. You can see how the logs look like in the tests: https://github.com/plausible/analytics/blob/v2.0.0/test/plausible_web/controllers/auth_controller/logs_test.exs

**Reply by ruslandoga** [reply]
Since the IP addresses are not logged, it's unlikely to be useful for fail2ban. And Plausible implements something similar to fail2ban for login actions [itself,](https://github.com/plausible/analytics/blob/0621fc0a77cc50add00baddbdd59f378324da483/test/plausible_web/controllers/auth_controller/logs_test.exs#L30-L49) so there is probably no need for an external tool.

**Reply by ruslandoga** [reply]
Here's the discussion where the feature was requested: https://github.com/plausible/analytics/discussions/2860

You can quickly skim it to get the reasoning behind it.

**Reply by sideskift** [reply]
I can see in the code that it has some kind of "Too Many Login Attempts" feature, but it will not block at firewall level. I think it would be nicer to have login attepts logged with IP so you could use fail2ban or even better CrowdSec to community block brute force attempts. 

And the IP informaion will be there because it is login in plausible and not websites linked to the servers, so nginx or the proxy web server, will know the IP

**Reply by ruslandoga** [reply]
OK, so if we add the IP addresses to the log messages, how would this integration work?

**Reply by sideskift** [reply]
CrowdSec scans a log file for bruteforce logins like repeated failed logins. Then you can define a bouncer process to act on it and it will then create the IP blocking rules in the firewall as well as sharing the attacker IP with the "cloud", other servers that run CrowdSec can then in advanced block that IP Address. In other words the community shares IP blocking firewall rules, no matter if it is an attack on software x or y. 

[See the docs for container log files and others](https://docs.crowdsec.net/docs/data_sources/docker)

**Reply by ruslandoga** [reply]
What's the expected format for the log lines containing the IP addresses?

**Reply by sideskift** [reply]
I am not an expert on crowdsec, but as I understand it there is not a specific file format. Instead you create parsers, and there are basically loads of parsers which scans the logfiles. So you could either adopt an existing format for login attepts logs, or create a parser for your own format. 

[More about crowdsec paraser patterns](https://docs.crowdsec.net/docs/parsers/patterns)

**Reply by ruslandoga** [reply]
In that case it might be safer to use the logs from a reverse proxy, like NGINX, and match on `"POST /login HTTP/1.1" 200` since we can't be certain right now that Plausible logs wouldn't produce false positives -- like maybe there is a log line in Plausible that is unrelated to auth but still matches the parser's format.

**Reply by sideskift** [reply]
But will you be able to see failed login attempts to Plausible from an NGINX proxy? I have tried to login with a nonsense account, and searched the NGINX log files, and I did not find any registration of a failed login in the NGINX log files.

**Reply by ruslandoga** [reply]
Yes, `<...> "POST /login HTTP/1.1" 200 <...>` is a failed login attempt. On success the request is redirected (status_code=302).

See https://github.com/plausible/analytics/discussions/2860#discussioncomment-5659071 for the full log lines.
